docs/add-runtime.md

```diff
@@ -0,0 +1,254 @@
+
+### 2.3 Writing the runtime
+Internally the runtime can be anything you like as long as it follows the standards set by the other runtimes.
+
+The best way to go about writing a runtime is like so:
+Initialize a web server that runs on port 3000 and uses any IP Address (0.0.0.0) and on each `POST` request do the following:
+1. Check that the `x-internal-challenge` header matches the `INTERNAL_RUNTIME_KEY` environment variable. If not return an error with a `401` status code and an `unauthorized` error message.
+2. Decode the executor's JSON POST request. This normally looks like so:
+```json
+{
+ "path": "/usr/code",
+ "file": "index.js",
+ "env": {
+         "hello":"world!"
+    },
+ "payload":"An Example Payload",
+ "timeout": 10
+}
+```
+For a compiled language you can disregard the `path` and `file` attribute if you like,
+
+`timeout` is also an optional parameter to deal with, if you can handle it please do. Otherwise, it doesn't matter since the connection will simply be dropped by the executor.
+
+You must create two classes for users to use within their scripts. A `Request` Class and a `Response` class
+The `Request` class must store `env`, `payload` and `headers` and pass them to the user's function. 
+The Request always goes before the response in the user's function parameters.
+The `Response` class must have two functions. 
+- A `send(string)` function which will return text to the request
+- and a `json(object)` function which will return JSON to the request setting the appropriate headers
+
+For interpreted languages use the `path` and `file` parameters to find the file and require it.
+Please make sure to add appropriate checks to make sure the imported file is a function that you can execute.
+
+5. Finally execute the function and handle whatever response the user's code returns. Try to wrap the function into a `try catch` statement to handle any errors the user's function encounters and return them cleanly to the executor with the error schema.
+
+### 2.4 The Error Schema
+All errors that occur during the execution of a user's function **MUST** be returned using this JSON Object otherwise OpenRuntimes will be unable to parse them for the user.
+```json
+{
+    "code": 500, // (Int) Use 404 if function not found or use 401 if the x-internal-challenge check failed.
+    "message": "Error: Tried to divide by 0 \n /usr/code/index.js:80:7", // (String) Try to return a stacktrace and detailed error message if possible. This is shown to the user.
+}
+```
+
+### 2.5 Writing your Dockerfile
+The Dockerfile is very important as it's the environment you are creating to run build the runtime and also run the code if you are writing an Interpreted Runtime (Compiled runtimes will use an alpine or ubuntu image to run the compiled executable.)
```

Comment:
```suggestion
The Dockerfile is very important as it's the environment you are creating to run build the runtime and also run the code if you are writing an interpreted runtime (compiled runtimes will use an `alpine` or `ubuntu` image to run the compiled executable.)
```

I think you should stick to using lower case for runtime, interpreted, compiled across the guide. 
When used in `Open Runtimes`, it should be capitalised